src/basecamp.cpp

```diff
@@ -657,20 +664,44 @@ std::list<item> basecamp::use_charges( const itype_id &fake_id, int &quantity )
     }
     return ret;
 }
+void basecamp::form_storage_zones( map &here, const tripoint_abs_ms &abspos )
+{
+    zone_manager &mgr = zone_manager::get_manager();
+    if( here.check_vehicle_zones( here.get_abs_sub().z() ) ) {
+        mgr.cache_vzones();
+    }
+    tripoint src_loc = here.getlocal( bb_pos ) + point_north;
+    if( mgr.has_near( zone_type_CAMP_STORAGE, abspos, 60 ) ) {
+        const std::vector<const zone_data *> zones = mgr.get_near_zones( zone_type_CAMP_STORAGE, abspos,
+                60 );
+        // Find the nearest unsorted zone to dump objects at
+        if( !zones.empty() ) {
+            if( zones != storage_zones ) {
```

Comment:
This fails to detect if the storage zone has been edited (same pointer, so the vectors of pointers compare equal). Fix incoming.